modules/core/src/ocl.cpp

```diff
@@ -3036,6 +3035,23 @@ Context Context::create(const std::string& configuration)
     return ctx;
 }
 
+void* Context::getProperty(long propertyId) const {
+    if (p == NULL)
+        return nullptr;
+    ::size_t size = 0;
+    if (CL_SUCCESS == clGetContextInfo(p->handle, CL_CONTEXT_PROPERTIES, 0, NULL, &size)) {
+        std::vector<cl_context_properties> prop(size / sizeof(cl_context_properties));
+        if (CL_SUCCESS == clGetContextInfo(p->handle, CL_CONTEXT_PROPERTIES, size, prop.data(), NULL)) {
```

Comment:
Just a question: can we cache this properties array? or its contents can be changed in runtime

Reply:
OpenCL doesn't allow to change properties (I would really like to have such ability - change media context property after context creation;)
But properties cached by OpenCL user-mode library, so these these two calls should work very fast.
Is it ok, or better cache on OpenCV side?

Reply:
OpenCL doesn't allow to change properties (I would really like to have such ability - change media context property after context creation;)
But properties cached by OpenCL user-mode library, so these these two calls should work very fast.
Is it ok, or better cache on OpenCV side?

Reply:
Thanks for information. Caching support here is low priority for now.